react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      image: 1
+    };
+    this.handleIndicatorClick = this.handleIndicatorClick.bind(this);
+    this.handleArrowClick = this.handleArrowClick.bind(this);
+    this.rotate = this.rotate.bind(this);
+    this.interval = setInterval(this.rotate, 3000);
+  }
+
+  handleIndicatorClick(e) {
+    let id = e.target.getAttribute('id');
+    id = parseInt(id, 10);
+    this.setState({
+      image: id
+    });
+  }
+
+  handleArrowClick(e) {
+    const id = e.target.getAttribute('id');
+    let position = this.state.image;
+    if (id === 'right') {
+      this.rotate();
+    } else if (id === 'left') {
+      if (position === 1) {
+        position = 5;
```

Comment:
Similar to `rotate`, you want to base your upper bound on the length of your `props.images`.